sycl/test/esimd/simd_replicate_deprecated.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: %clangxx -fsycl -fsyntax-only -Xclang -verify %s
+
+// This test checks deprecated replicate ESIMD APIs
+
+#include <sycl/ext/intel/experimental/esimd.hpp>
+
+using namespace sycl::ext::intel::experimental::esimd;
+
+bool test_replicate1() {
+  simd<int, 8> v0(0, 1);
+  // expected-warning@+3 2 {{deprecated}}
+  // expected-note@sycl/ext/intel/experimental/esimd/esimd.hpp:347 {{}}
```

Comment:
This would simplify test maintenance a lot w/o really compromising test strength.
```suggestion
  // expected-note@sycl/ext/intel/experimental/esimd/esimd.hpp:* {{}}
```
BTW, I had lots of troubles (need to fix lines etc) with copy_to/copy from, so that I even doubt now it is worth it to add deprecation tests any more.

Reply:
I think there is at least some value in having those tests as they ensure that the previous APIs still work. And I hope soon we will get rid of deprecated APIs as well as the tests.
I will update the test as you suggested.